src/createRouter.ts

```diff
@@ -121,6 +143,21 @@ const createRouter: RouterFactory = compose<RouterMixin, RouterOptions>({
 		}
 	},
 
+	observeHistory(history: History, context: Context, dispatchInitial: boolean): PausableHandle {
+		if (!historyMap.has(this)) {
```

Comment:
I'd prefer an early return (or throw) if `historyMap.has(this)` so the bulk of the function doesn't have to be indented.